core/src/main/java/org/dcache/nfs/vfs/PseudoFs.java

```diff
@@ -65,17 +69,23 @@
     private final VirtualFileSystem _inner;
     private final ExportFile _exportFile;
     private final RpcAuth _auth;
+    private final sattr3 _attr;
 
     private final static int ACCESS4_MASK =
             ACCESS4_DELETE | ACCESS4_EXECUTE | ACCESS4_EXTEND
             | ACCESS4_LOOKUP | ACCESS4_MODIFY | ACCESS4_READ;
 
     public PseudoFs(VirtualFileSystem inner, RpcCall call, ExportFile exportFile) {
+        this(inner, call, exportFile, null);
+    }
+
+    public PseudoFs(VirtualFileSystem inner, RpcCall call, ExportFile exportFile, sattr3 attr) {
```

Comment:
I don't like the first option because it seems like a work-around instead of a fix. Second option seems to be too much work for me at this time. I went for a third option by simply replacing the v3 attributes with a translated Stat, have a look and tell me what you think. If this is no good I'll go back to the first option you suggested,.
The client is our (Primarydata's) metadata server doing nfsv3 rpc calls to nfsv3 servers.